sound/soc/intel/boards/sdw_rt711_rt1308_rt715.c

```diff
@@ -316,16 +316,39 @@ static const struct snd_kcontrol_new controls[] = {
 	SOC_DAPM_PIN_SWITCH("Speaker"),
 };
 
+static int set_amp_tdm_slots(struct snd_soc_pcm_runtime *rtd, bool tdm_mode)
+{
+	struct snd_soc_dai *codec_dai;
+	int ret = 0;
+	int val;
+	int i;
+
+	for_each_rtd_codec_dai(rtd, i, codec_dai) {
+		if (tdm_mode)
+			/* tx_mask 0, rx_mask with only one bit, single slot) */
+			val = snd_soc_dai_set_tdm_slot(codec_dai, 0, 1 << i, 1, 0);
```

Comment:
tx_mask or rx_mask ? As I know, tx_mask is for playback, rx_mask is for capture

Reply:
Ah, that's a good one. I think I use the same convention in the rt1308 and machine driver, so things should work but I will have to check if this is correct or not.

Reply:
@RanderWang tx_mask is used when the DAI transmits data, and rx_mask when it consumes data on the bus.

So for a codec_dai, in a playback scenario it's rx_mask that should be used.